src/Core/Engine/Systems/LightManager.cpp

```
@@ -1,23 +1,65 @@
+    };
+}
+
+LightManager::LightManager(entt::registry& scene_registry)
+    : m_SceneRegistry(scene_registry)
+{
+    scene_registry.on_construct<PointLight>().connect<&LightManager::OnPointLightChanged>(*this);
+    scene_registry.on_destroy<PointLight>().connect<&LightManager::OnPointLightChanged>(*this);
+    scene_registry.on_update<PointLight>().connect<&LightManager::OnPointLightChanged>(*this);
+
+    scene_registry.on_construct<SpotLight>().connect<&LightManager::OnSpotlightChanged>(*this);
+    scene_registry.on_destroy<SpotLight>().connect<&LightManager::OnSpotlightChanged>(*this);
```

Cppcheck (2.20.0)

[style] The function 'GetLight' is never used.

(unusedFunction)